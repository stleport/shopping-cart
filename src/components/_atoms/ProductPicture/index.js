import React from 'react';
import PropTypes from 'prop-types';

const ProductPicture = ({ altText, source }) => (
  (
    <img 
      className="image" 
      alt={altText} 
      src={source} 
      onError={(e) => { e.target.onerror = null; e.target.src = 'frichti.jpg'; }}
    />
  )
);

ProductPicture.propTypes = {
  altText: PropTypes.string.isRequired,
  source: PropTypes.string.isRequired
};

export default ProductPicture;
