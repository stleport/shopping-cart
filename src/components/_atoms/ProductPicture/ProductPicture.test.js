import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ProductPicture from '.';

function ProductPictureSetup(newProps = {}) {
  const props = {
    altText: 'Alt text',
    source: 'http://example.com',
  };

  const ProductPictureJSX = (
    <ProductPicture {...props} />
  );

  const ProductPictureComponent = shallow(ProductPictureJSX);
  return {
    ProductPictureComponent,
    ProductPictureJSX,
    props
  };
}

describe('ProductPicture', () => {

  it('renders correctly ProductPicture component', () => {
    const { ProductPictureJSX } = ProductPictureSetup();
    const ProductPictureComponent = renderer.create(ProductPictureJSX).toJSON();
    expect(ProductPictureComponent).toMatchSnapshot();
  });

});
