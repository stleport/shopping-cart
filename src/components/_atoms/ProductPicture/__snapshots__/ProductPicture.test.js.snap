// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPicture renders correctly ProductPicture component 1`] = `
<img
  alt="Alt text"
  className="image"
  onError={[Function]}
  src="http://example.com"
/>
`;
